### DISCUSSION TITLE
Win10 IoT installation crashes BSOD (DRIVER_VERIFIER_DMA_VIOLATION)

### DISCUSSION THREAD

**Discussion by DigCamara** [ipxe/ipxe#855] [Q&A] [answered]
I am using wimboot to install a Windows 10 IoT LTSC version, pretty much like it is described [here](https://ipxe.org/wimboot?s[]=wimboot). 

It works; however, once the iPXE hands off to the Windows boot process, a DRIVER_VERIFIER_DMA_VIOLATION crashes the install and restarts the process. Sometimes it takes one, sometimes several restarts to continue the installation process and from then on there are no problems.

I have used the same Windows ISO to generate a USB bootable disk and it works without a hitch, so I am guessing there is some problem with the transition between iPXE and the actual Setup. I was wondering if there is some extended way to debug the issue.

**Comment by mcb30** [comment, marked as answer]
We have seen a similar issue before, which was fixed in commit https://github.com/ipxe/ipxe/commit/562c74e1e.  Please make sure that you are using an up-to-date version of iPXE.

Which git commit of iPXE are you using, and what do you get from the `ifstat` command?

**Reply by DigCamara** [reply]
You are right, that sounds exactly like my problem.  Since I didn't build the ipx.efi myself, I will have to verify the commit (the ipx.efi file was generated around 2022.05). I also cannot physically access the machine until Monday, but I will keep you posted.

**Reply by DigCamara** [reply]
Update: I tried using the [ipxe.efi](https://boot.ipxe.org/ipxe.efi) file at https://boot.ipxe.org/ and it still fails. `ifstat` reports one link up via the MAC address I expected, via rte18168, 3 closed: a secondary network card, not connected and the expected MAC address via SNP. 

The connected link reports unreachable network to ipxe.org, which would be normal, since the equipment is not connected to internet.

I will try to find a Linux machine that allows me to compile the latest version on site.

**Reply by mcb30** [reply]
Could you please post a screenshot (or transcript) of the exact output from the `ifstat` command?  This is generally more useful than an attempt to summarise what you think it is telling you. 🙂

**Reply by DigCamara** [reply]
I hope this helps 

![grafik](https://user-images.githubusercontent.com/1839211/214003703-4e43024e-ec2f-4638-a10e-98fd35dd8fcc.png)

**Reply by DigCamara** [reply]
Update: I compiled the version based on commit [8f59911b2](https://github.com/ipxe/ipxe/commit/8f59911b2) and still have the same issue

**Reply by mcb30** [reply]
Thank you for the screenshot.  This immediately shows a problem: you should definitely not have the duplicate NICs `net1` and `net3` showing up.  This may or may not be the cause of your problem, but it's definitely something going wrong before your problem occurs and therefore something that should be fixed.

I have a suspicion as to the root cause: will do some testing.

**Reply by mcb30** [reply]
> Thank you for the screenshot. This immediately shows a problem: you should definitely not have the duplicate NICs `net1` and `net3` showing up. This may or may not be the cause of your problem, but it's definitely something going wrong before your problem occurs and therefore something that should be fixed.

@DigCamara This problem should (I hope) be fixed in commit https://github.com/ipxe/ipxe/commit/68734b9a4.  Could you please pull the latest code and retest?

**Reply by DigCamara** [reply]
Preliminary update: I pulled the latest code and have so far avoided any BSOD. I will continue testing, however I can confirm `net1`: and `net3:` are not appearing anymore

**Reply by mcb30** [reply]
That's great: at least one issue is now solved. :slightly_smiling_face:

**Reply by DigCamara** [reply]
I have done around 10 test runs so far, all without BSOD. Since I usually could not get 3 runs in a row without BSOD, I think the issue has been resolved. I will continue testing; should anything come up I will keep you posted.

**Reply by mcb30** [reply]
That's good to know: thank you for reporting back!
